packages/react-components/lib/dispensers/dispenser-accordion.tsx

```diff
@@ -91,20 +114,21 @@ export const DispenserAccordion = React.forwardRef(
             variant: statusLabelClass ? 'normal' : 'unknown',
           }}
         />
-        {dispenserState ? (
-          <ItemAccordionDetails>
-            <DispenserInfo dispenser={dispenserState} />
-          </ItemAccordionDetails>
-        ) : (
-          <React.Fragment>
-            <Divider />
-            <Typography className={classes.typography} variant="body1">
-              The state of <b>{dispenser}</b> dispenser is unknown. Please check if{' '}
-              <b>{dispenser} </b>
-              is working properly.
-            </Typography>
-          </React.Fragment>
-        )}
+        <ItemAccordionDetails>
+          <ErrorOverlay
+            errorMsg={
+              !dispenserState
+                ? 'Dispenser is not sending states. Please check if it is working properly.'
+                : undefined
+            }
+          >
+            {dispenserState ? (
+              <DispenserInfo dispenser={dispenserState} />
+            ) : previousState ? (
+              <DispenserInfo dispenser={previousState} overrideStyle={overrideClasses} />
+            ) : null}
```

Comment:
just a nit, and it's up to you, an alternative to avoid nested ternary operators can be this

```
{dispenserState && <DispenserInfo dispenser={dispenserState} />}
{(!dispenserState && previousState) && <DispenserInfo dispenser={previousState} overrideStyle={overrideClasses}}
```

If you think this logic is convoluted you can leave it as it is.